layouts/partials/head.html

```diff
@@ -13,6 +13,14 @@
 <link rel="alternate" hreflang="{{ .Language.Lang }}" href="{{ .Permalink }}">
 {{ end -}}
 
+<!-- lead search engine from staging site to main domain -->
+<link x-hugo="RelPermalink" rel="canonical" href="https://kubernetes.io{{ .RelPermalink }}">
+<link x-hugo="RelPermalink-absURL" rel="canonical" href="{{ .RelPermalink | absURL }}">
+<link x-hugo="Permalink" rel="canonical" href="{{ .Permalink }}">
+{{ if hugo.IsProduction -}}
+<link x-hugo="Permalink-IsProd" rel="canonical" href="{{ .Permalink }}">
+{{ end -}}
```

Comment:
Just use `.Permalink` and `| absURL` for test.

The current result is:
```html
<link x-hugo="RelPermalink" rel="canonical" href="https://kubernetes.io/">
<link x-hugo="RelPermalink-absURL" rel="canonical" href="https://deploy-preview-38932--kubernetes-io-main-staging.netlify.app/">
<link x-hugo="Permalink" rel="canonical" href="https://deploy-preview-38932--kubernetes-io-main-staging.netlify.app/">
```

I think we have to hard-code the `https://kubernetes.io` link or it will be meaningless for this scenario.

Reply:
> I think we have to hard-code the `https://kubernetes.io` link or it will be meaningless for this scenario.

The only site we need to correct - that I know of - is https://main--kubernetes-io-main-staging.netlify.app/

If that site sets the hostname to `main--kubernetes-io-main-staging.netlify.app` then we can detect that and render the site specially (for example: exclude robot crawling using `robots.txt`). If that site sets the hostname to `kubernetes.io` then we don't need to hard code it.

Also, hard coding **will** break the v1.26 release docs as soon as we do the v1.27 release.
We definitely shouldn't hard code the hostname here without extra measures to avoid that problem.